app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Stepper.kt

```diff
@@ -0,0 +1,693 @@
+      @Composable
+      @ReadOnlyComposable
+      get() = MediumSizeLabeledStepperTokens.LabelTextFont.value
+    override val shape: Shape
+      @Composable
+      @ReadOnlyComposable
+      get() = MediumSizeDefaultStepperTokens.ContainerShape.value
+  }
+
+  data object Small : StepperSize() {
+    override val defaultContentPadding: PaddingValues
+      get() = PaddingValues(
+        top = SmallSizeDefaultStepperTokens.TopPadding,
+        bottom = SmallSizeDefaultStepperTokens.BottomPadding,
+        start = SmallSizeDefaultStepperTokens.HorizontalPadding,
+        end = SmallSizeDefaultStepperTokens.HorizontalPadding,
+      )
+    override val labelContentPadding: PaddingValues
+      get() = PaddingValues(
+        top = SmallSizeLabeledStepperTokens.TopPadding,
+        bottom = SmallSizeLabeledStepperTokens.BottomPadding,
+        start = SmallSizeLabeledStepperTokens.HorizontalPadding,
+        end = SmallSizeLabeledStepperTokens.HorizontalPadding,
+      )
+    override val symbolsPadding: PaddingValues
+      get() = PaddingValues(
+        top = SmallSizeDefaultStepperTokens.StepperTopPadding,
+        bottom = SmallSizeDefaultStepperTokens.StepperBottomPadding,
+      )
+    override val labelStyleErrorTextPadding: PaddingValues
+      get() = PaddingValues(
+        start = SmallSizeLabeledStepperTokens.ErrorTextStartPadding,
+        end = SmallSizeLabeledStepperTokens.ErrorTextEndPadding,
+      )
+    override val defaultStyleErrorTextPadding: PaddingValues
+      get() = PaddingValues(
+        start = SmallSizeDefaultStepperTokens.ErrorTextStartPadding,
+        end = SmallSizeDefaultStepperTokens.ErrorTextEndPadding,
+      )
+    override val defaultMinHeight: Dp
+      get() = SmallSizeDefaultStepperTokens.MinHeight
+    override val labelMinHeight: Dp
+      get() = SmallSizeLabeledStepperTokens.MinHeight
+    override val defaultCommonTextStyle: TextStyle
+      @Composable
+      @ReadOnlyComposable
+      get() = SmallSizeDefaultStepperTokens.TextFont.value
+    override val labeledCommonTextStyle: TextStyle
+      @Composable
+      @ReadOnlyComposable
+      get() = SmallSizeLabeledStepperTokens.TextFont.value
+    override val labelTextStyle: TextStyle
+      @Composable
+      @ReadOnlyComposable
+      get() = SmallSizeLabeledStepperTokens.LabelTextFont.value
+    override val shape: Shape
+      @Composable
+      @ReadOnlyComposable
+      get() = SmallSizeDefaultStepperTokens.ContainerShape.value
+  }
+}
+
+@Immutable
+private data class StepperColors(
+  private val containerColor: Color,
+  private val labelColor: Color,
+  private val textColor: Color,
+  private val enabledSymbolColor: Color,
+  private val disabledSymbolColor: Color,
+  private val pulsatingContainerColor: Color,
+  private val pulsatingContentColor: Color,
+  val errorTextColor: Color,
+) {
+  @Stable
+  fun containerColor(pulsating: Boolean): Color = when {
+    pulsating -> pulsatingContainerColor
+    else -> containerColor
+  }
```

Comment:
Yeah so for the animated colors/values, can you try to see if you could make it work by adding the animations here instead? The exact same way I mentioned in the other PR, like we do here for example https://github.com/HedvigInsurance/android/blob/b5bf4698ac78da1f9753cb9e834467901210caaf/app/core/core-design-system/src/main/kotlin/com/hedvig/android/core/designsystem/component/navigation/HedvigNavigationBarItem.kt#L320-L331

This would avoid the code which starts adding up quite a lot at the beginning of the component with having to manage all the animated values, especially when it's a lot of them.

Lmk if what I am saying does not make sense.